src/main/java/com/yugabyte/sample/apps/SqlUpdatesWithProcedures.java

```diff
@@ -0,0 +1,275 @@
+  static {
+    // Disable the read-write percentage.
+    appConfig.readIOPSPercentage = -1;
+    // Set the read and write threads to 1 each.
+    appConfig.numReaderThreads = 1;
+    appConfig.numWriterThreads = 1;
+    // The number of keys to read.
+    appConfig.numKeysToRead = -1;
+    // The number of keys to write. This is the combined total number of inserts and updates.
+    appConfig.numKeysToWrite = -1;
+    // The number of unique keys to write. This determines the number of inserts (as opposed to
+    // updates).
+    appConfig.numUniqueKeysToWrite = NUM_UNIQUE_KEYS;
+  }
+
+  // The default table name to create and use for CRUD ops.
+  private static final String DEFAULT_TABLE_NAME = "PostgresqlKeyValue";
+
+  // The shared prepared select statement for fetching the data.
+  private volatile PreparedStatement preparedSelect = null;
+
+  // The shared prepared insert statement for inserting the data.
+  private volatile PreparedStatement preparedInsert = null;
+
+  // Lock for initializing prepared statement objects.
+  private static final Object prepareInitLock = new Object();
+
+  public SqlUpdatesWithProcedures() {
+    buffer = new byte[appConfig.valueSize];
+  }
+
+  @Override
+  public void createTablesIfNeeded() throws Exception {
+    // Check that (extra_ required options are set (maxWrittenKey and loadTesterUUID) set.
+    if (appConfig.maxWrittenKey <= 0) {
+      LOG.fatal("Workload requires option --max_written_key to be set. \n " +
+                    "Run 'java -jar yb-sample-apps.jar --help SqlUpdatesWithProcedures' for more details");
+      System.exit(1);
+    }
+    if (CmdLineOpts.loadTesterUUID == null) {
+      LOG.fatal("Workload requires option --uuid to be set. \n " +
+                    "Run 'java -jar yb-sample-apps.jar --help SqlUpdatesWithProcedures' for more details");
+      System.exit(1);
+    }
+
+    // Expecting that the SqlInserts ap was already run and created the table.
+    // This app just reuses the table and updates the values for the existing rows.
+    Connection connection = getPostgresConnection();
+    DatabaseMetaData dbm = connection.getMetaData();
+    ResultSet tables = dbm.getTables(null, null, getTableName(), null);
+    if (!tables.next()) {
+      LOG.fatal(
+          String.format("Could not find the %s table. Did you run SqlInserts first?" +
+                            "Run 'java -jar yb-sample-apps.jar --help SqlUpdatesWithProcedures' for more details",
+                        getTableName()));
+      System.exit(1);
+    }
+  }
+
+  public String getTableName() {
+    String tableName = appConfig.tableName != null ? appConfig.tableName : DEFAULT_TABLE_NAME;
+    return tableName.toLowerCase();
+  }
+
+  @Override
+  public void createStoredProcedures() throws Exception {
+    // Drop previous procedures.
+    getPostgresConnection().createStatement().execute(
+      String.format("DROP PROCEDURE IF EXISTS %s;", getStoredProcedureName()));
```

Comment:
Allowing the Procedure name to be named doesn't add anything.
Can't we use the default name all the time?

Reply:
I feel like it's similar to tableName, which we do have in appConfig as well.